packages/aws-cdk-lib/custom-resource-config/test/custom-resource-config.test.ts

```diff
@@ -0,0 +1,547 @@
+import { Capture, Template, Match } from '../../assertions';
+import * as cdk from '../../core'
+import * as logs from '../../aws-logs';
+import * as s3 from '../../aws-s3';
+import * as ses from '../../aws-ses';
+import * as s3deploy from '../../aws-s3-deployment';
+import * as events from "../../aws-events";
+import * as targets from "../../aws-events-targets";
+import * as iam from "../../aws-iam";
+import { CustomResourceConfig } from '../custom-resource-config';
+
+describe('CustomResourceConfig empty retention', ()=> {
+    test('catch thrown error if no retention', ()=> {
+        const app = new cdk.App();
+        try {
+            CustomResourceConfig.of(app).addLogRetentionLifetime()
+        } catch (e) {
+            expect(e.message).toBe("Retention must have a value");
+        }
+    })
+})
+describe('aws-ses drop spams', () => {
+    test('Drop spams - UnModified by Aspects', () => {
+        const app = new cdk.App();
+        const stack = new cdk.Stack(app);
+
+        new ses.ReceiptRuleSet(stack, 'RuleSet', {
+            dropSpam: true,
+        });
+
+        const template = Template.fromStack(stack);
+
+        template.resourceCountIs('AWS::Logs::LogGroup', 0);
+    })
+    test('Drop spams - Modified by Aspects', () => {
+        const SET_LOG_RETENTION = logs.RetentionDays.TEN_YEARS;
+
+        const app = new cdk.App();
+        CustomResourceConfig.of(app).addLogRetentionLifetime(SET_LOG_RETENTION);
+        const stack = new cdk.Stack(app);
+
+        new ses.ReceiptRuleSet(stack, 'RuleSet', {
+            dropSpam: true,
+        });
+
+        const template = Template.fromStack(stack);
+
+        template.resourceCountIs('AWS::Logs::LogGroup', 1);
+        template.hasResourceProperties('AWS::Logs::LogGroup', {
+            RetentionInDays: SET_LOG_RETENTION
+        })
+        template.hasResourceProperties('AWS::Lambda::Function', {
+            LoggingConfig: {
+                LogGroup: {
+                    Ref: Match.stringLikeRegexp('SingletonLambda')
+                }
+            }
+        })
+    })
+    test('Drop spams - snapshot', () => {
+        const SET_LOG_RETENTION = logs.RetentionDays.TEN_YEARS;
+
+        const app = new cdk.App();
+        CustomResourceConfig.of(app).addLogRetentionLifetime(SET_LOG_RETENTION);
+        const stack = new cdk.Stack(app);
+
+        new ses.ReceiptRuleSet(stack, 'RuleSet', {
+            dropSpam: true,
+        });
+
+        const template = Template.fromStack(stack);
+
+        expect(template.toJSON()).toMatchSnapshot();
+    })
+})
+describe('aws-events-targets awsapi', () => {
```

Comment:
I appreciate the pains you've taken to do this for all these different custom resources, but this doesn't add much to the testing.

You only need to test the different cases that impact your implementation. In this case, there's only two types of custom resources that matter; those backed by Singleton lambdas and those backed by non-Singleton lambdas. 

Choose one of each and use them throughout the entire test file. In the test string, don't refer to them by name; eg, there's nothing gained by saying "S3Deployment lambda log retention set to blah". Instead, say, "singleton-backed custom resource logging configuration set to blah", or something along those lines. That communicates the case that you care about, rather than the arbitrary choice of custom resource.